### PR TITLE
Add gnome-text-editor application in business-vm

### PR DESCRIPTION
**Pull request by Nerox9**
## Description of changes

gnome-text-editor is added into business-vm with icon

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [ ] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [ ] Change can be updated with `nixos-rebuild ... switch`

## Instructions for Testing

- [ ] List all targets that this applies to:
  - [ ] Lenovo X1 `x86_64`
- [x] Is this a new feature
  - [ ] List the test steps to verify:
     - [ ] Open the ghaf launcher on the left down corner in the guivm.
    - [ ] Run the gnome-text-editor app by clicking the icon.
- [ ] If it is an improvement how does it impact existing functionality?

### CODE REVIEW

File: modules/desktop/graphics/labwc.config.nix [hunk 1]
```diff
@@ -298,6 +298,12 @@ in
         path = "${pkgs.labwc}/bin/labwc --exit";
         icon = "${pkgs.icon-pack}/system-log-out.svg";
       }
+      {
+        name = "Notepad Next";
+        description = "Notepad Application";
+        path = "${pkgs.notepad-next}/bin/NotepadNext";
```
Comment (vunnyso) [inline modules/desktop/graphics/labwc.config.nix:304]: Not supported on `Aarch64` https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/editors/notepad-next/default.nix#L41
And run-checks are failing, please fix that.

Comment (Nerox9) [reply modules/desktop/graphics/labwc.config.nix]: It is moved from launchers of labwc config to desktop service.

**Comment by milva-unikie** [comment]
**Tested on Lenovo-X1** (`nixos-rebuild switch`)
- Notepad opens
- Text files can be created, opened, edited and saved
- All of the options I clicked through are working as expected

**Comment by Nerox9** [comment]
> How about an application like gnome-text-editor, which would be friendly to a non-technical user and we wouldn’t be bringing in Qt dependencies.
> 
> Also readme of the notepad-next project states:
> 
> > it should not be considered safe for critically important work

Notepad-Next is replaced by gnome-text-editor app.

**Comment by milva-unikie** [comment]
**Tested on Lenovo-X1** (`nixos-rebuild switch`)
- Gnome Text Editor opens.
- Text files can be created, opened, edited and saved.
- This is much more simplified than Notepad Next, but I think it does everything needed.
- Some options are not working in menus:
  - In Main Menu 'Dark Style' and 'Light Style' do nothing.
  - In Options Menu (the one left of main menu) only document type can be changed. All the other options seem to do nothing. 
  - -> Probably not the end of the world if some of these don't work, but I would guess they should be working in the app?

**Review by vunnyso — APPROVED** [review]

**Comment by milva-unikie** [comment]
**Tested on Lenovo-X1** (`nixos-rebuild switch`)
- Gnome Text Editor opens in business-vm.
- Text files can be created, opened, edited and saved inside business-vm.
- Green line around the window indicates that it is running in business-vm.
- This is much more simplified than Notepad Next, but I think it does everything needed.

Issues:
- This app does not have the standard colorful border (with minimize, maximize and close buttons). All the other apps have it in business-vm.  This should probably be fixed before merging.
- As a smaller issue: Some options are not working in menus
  - In Main Menu 'Dark Style' and 'Light Style' do nothing.
  - In Options Menu (the one left of main menu) only document type can be changed. All the other options seem to do nothing. 
  - -> Probably not the end of the world if some of these don't work, but I would guess they should be working in the app?

**Comment by Nerox9** [comment]
> **Tested on Lenovo-X1** (`nixos-rebuild switch`)
> 
>     * Gnome Text Editor opens in business-vm.
> 
>     * Text files can be created, opened, edited and saved inside business-vm.
> 
>     * Green line around the window indicates that it is running in business-vm.
> 
>     * This is much more simplified than Notepad Next, but I think it does everything needed.
> 
> 
> Issues:
> 
>     * This app does not have the standard colorful border (with minimize, maximize and close buttons). All the other apps have it in business-vm.  This should probably be fixed before merging.
> 
>     * As a smaller issue: Some options are not working in menus
>       
>       * In Main Menu 'Dark Style' and 'Light Style' do nothing.
>       * In Options Menu (the one left of main menu) only document type can be changed. All the other options seem to do nothing.
>       * -> Probably not the end of the world if some of these don't work, but I would guess they should be working in the app?

It has a similar appearance to AppFlowy as @humaidq-tii mentioned. after the new commit.
Missing icons and not working UI are fixed by dconf.

**Comment by milva-unikie** [comment]
**Tested on Lenovo-X1** (`nixos-rebuild switch`)
- Gnome Text Editor opens in business-vm.
- Text files can be created, opened, edited and saved inside business-vm.
- Green line around the window indicates that it is running in business-vm.
- The app has the standard Ghaf border (with minimize, maximize and close buttons).
- All menu options now work.

Can I make one last small request? The color of the border is quite aggressively green and the contrast in the top bar is not good. As far as I know we don't have guidelines on what the colors should be, but something a bit softer would be great.

Sorry if I was unclear about that in my previous comment. The apps in business-vm have two "borders". The green inside line indicating that the app is in business-vm and then the outside border that almost all apps have.

image

**Review by milva-unikie — COMMENTED** [review]
Color is fixed now!

**Review by brianmcgillion — APPROVED** [review]

**Pull request merged**